lib/rules/prefer-presence-queries.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+const { getDocsUrl } = require('../utils');
+
+const PRESENCE_MATCHERS = ['toBeInTheDocument', 'toBeTruthy', 'toBeDefined'];
+const ABSENCE_MATCHERS = ['toBeNull', 'toBeFalsy'];
+
+module.exports = {
+  meta: {
+    docs: {
+      category: 'Best Practices',
+      description:
+        'Disallow the use of `getBy*` queries when checking elements are not present',
```

Comment:
I would word this to be more generic
```suggestion
        'Ensure appropriate get*/query* queries are used with their respective matchers',
```
or something to that effect